### ISSUE TITLE
Move/rename package refactoring

### ISSUE THREAD

**Issue by muhmud** [emacs-lsp/lsp-java#178]
Would it be possible to implement move/rename package refactoring? This appears to be implemented in the Eclipse JDT server: https://github.com/eclipse/eclipse.jdt.ls/blob/master/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/corext/refactoring/rename/RenamePackageProcessor.java

For this implementation, would it also be possible to have the option of creating the new package if it does not exist? It would be great to have the same thing for moving a class to a new package, i.e. have the option to create a new package if one does not already exist, just like in Eclipse.

**Comment by ivanhjc** [comment]
Yeah I'm looking for this feature as well. Currently the only way I find that can do this is using Eclim's `M-x eclim-java-refactor-move-class`. See http://www.goldsborough.me/emacs,/java/2016/02/24/22-54-16-setting_up_emacs_for_java_development/#refactoring

**Comment by ericdallo** [comment]
Any news on that?

**Comment by chnnn** [comment]
It seems that for simple rename refactoring,  `lsp-rename` might do the job.
For complex usage, maybe use Eclipse's Alt+Shift+r ...
